### PR TITLE
[FEATURE-SERVERSIDE-APPLY] add userAgent info to create and update

### PR DESCRIPTION
**Pull request by kwiesmueller**

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/endpoints/handlers/patch.go [hunk 1]
```diff
@@ -551,7 +551,7 @@ func (p *patcher) patchResource(ctx context.Context, scope RequestScope) (runtim
 }
 
 func (p *patcher) prefixFromUserAgent() string {
-	return strings.Split(p.userAgent, "/")[0]
+	return prefixFromUserAgent(p.userAgent)
```
Comment (apelisse) [inline staging/src/k8s.io/apiserver/pkg/endpoints/handlers/patch.go]: I'd rather remove this `patcher` method and be consistent in all 3 locations

Comment (kwiesmueller) [reply staging/src/k8s.io/apiserver/pkg/endpoints/handlers/patch.go]: Okay, after you suggested to do it as a method in the last PR i kept it, but agreed.

**Review by apelisse — APPROVED** [review]
/lgtm
/approve

**Review by apelisse — APPROVED** [review]
/lgtm

**Pull request merged**